### PR TITLE
Convert controlplane files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/controlplane/controller/apiserverleasegc/gc_controller.go [hunk 1]
```diff
@@ -73,9 +73,9 @@ func NewAPIServerLeaseGC(clientset kubernetes.Interface, gcCheckPeriod time.Dura
 // Run starts one worker.
 func (c *Controller) Run(stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
-	defer klog.Infof("Shutting down apiserver lease garbage collector")
+	defer klog.InfoS("Shutting down API Server lease garbage collector")
```
Comment (cici37) [inline pkg/controlplane/controller/apiserverleasegc/gc_controller.go:76]: minor: We have `API Server`, `API server`, `apiserver` combined in code base. Most of the customer facing messages were using `API server`. e.g. https://github.com/kubernetes/kubernetes/blob/1c1527f5198b4e8391ce37082dbe166879750295/staging/src/k8s.io/client-go/tools/clientcmd/overrides.go#L180

Comment (scottilee) [reply pkg/controlplane/controller/apiserverleasegc/gc_controller.go:76]: Fixed to "API server".

**Comment by scottilee** [comment]
@cici37 Thanks for the review! I lowercased "server" and updated the .structured_logging file.

**Comment by cici37** [comment]
There are still files under `//pkg/controlplane` haven't converted to structured logs. You could either have all changes under `//pkg/controlplane` in this pr OR have a separate pr for the rest of files and update pr description and hold the deletion from `hack/.structured_logging` to make it clear. Thank you :)

File: hack/.structured_logging [hunk 2]
```diff
@@ -1,2 +1,3 @@
 cmd/kubelet/...
+pkg/controlplane/...
```
[the remark this answers is not in the document]
Comment (scottilee) [reply hack/.structured_logging:2]: I believe I have updated all files that need structured logging under pkg/controlplane/ but let me know if I've missed any.

[the remark this answers is not in the document]
Comment (cici37) [reply hack/.structured_logging:2]: FYI:
```/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go:440:2: unstructured logging function "Infof" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/reconcilers/instancecount.go:103:3: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/reconcilers/instancecount.go:140:2: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/reconcilers/lease.go:233:2: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/tunneler/ssh.go:115:3: unstructured logging function "Warning" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:91:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:161:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:165:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:193:3: unstructured logging function "Warning" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:288:5: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:271:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:275:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:293:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:307:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:594:4: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:612:5: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/services.go:37:3: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go:440:2: unstructured logging function "Infof" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/reconcilers/instancecount.go:103:3: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/reconcilers/instancecount.go:140:2: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/reconcilers/lease.go:233:2: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/tunneler/ssh.go:115:3: unstructured logging function "Warning" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:91:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:161:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:165:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:193:3: unstructured logging function "Warning" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/controller.go:288:5: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:271:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:275:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:293:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:307:3: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:594:4: unstructured logging function "Warningf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/instance.go:612:5: unstructured logging function "Fatalf" should not be used
/home/prow/go/src/k8s.io/kubernetes/pkg/controlplane/services.go:37:3: unstructured logging function "Warningf" should not be used
```

[the remark this answers is not in the document]
Comment (scottilee) [reply hack/.structured_logging:2]: @cici37 How did you get this list? What command did you run? Thanks.

[the remark this answers is not in the document]
Comment (cici37) [reply hack/.structured_logging:2]: You could run `./hack/verify-structured-logging.sh` locally(https://github.com/kubernetes/kubernetes/blob/master/hack/verify-structured-logging.sh). And you also could take a look at the result of presubmits by clicking `details` button under `pull-kubernetes-verify`. :)

[the remark this answers is not in the document]
Comment (scottilee) [reply hack/.structured_logging:2]: Thanks!

**Comment by scottilee** [comment]
> There are still files under `//pkg/controlplane` haven't converted to structured logs. You could either have all changes under `//pkg/controlplane` in this pr OR have a separate pr for the rest of files and update pr description and hold the deletion from `hack/.structured_logging` to make it clear. Thank you :)

@cici37 I got ambitious and tried to get them all :) Let me know if I'm missing anything else

File: pkg/controlplane/instance.go [hunk 3]
```diff
@@ -579,15 +584,15 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
 	for _, restStorageBuilder := range restStorageProviders {
 		groupName := restStorageBuilder.GroupName()
 		if !apiResourceConfigSource.AnyVersionForGroupEnabled(groupName) {
-			klog.V(1).Infof("Skipping disabled API group %q.", groupName)
+			klog.V(1).InfoS("Skipping disabled API group.", "groupName", groupName)
```
Comment (cici37) [inline pkg/controlplane/instance.go:587]: Maybe the key "apiGroup" would be more clear? Or at least keep it consistent.

File: pkg/controlplane/instance.go [hunk 4]
```diff
@@ -579,15 +584,15 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
...
 			continue
 		}
 		apiGroupInfo, enabled, err := restStorageBuilder.NewRESTStorage(apiResourceConfigSource, restOptionsGetter)
 		if err != nil {
 			return fmt.Errorf("problem initializing API group %q : %v", groupName, err)
 		}
 		if !enabled {
-			klog.Warningf("API group %q is not enabled, skipping.", groupName)
+			klog.InfoS("API group is not enabled, skipping.", "group", groupName)
```
Comment (cici37) [inline pkg/controlplane/instance.go:595]: Same here

File: pkg/controlplane/instance.go [hunk 5]
```diff
@@ -596,16 +601,17 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
 		// This is a spot above the construction of individual storage handlers so that no sig accidentally forgets to check.
 		resourceExpirationEvaluator.RemoveDeletedKinds(groupName, apiGroupInfo.Scheme, apiGroupInfo.VersionedResourcesStorageMap)
 		if len(apiGroupInfo.VersionedResourcesStorageMap) == 0 {
-			klog.V(1).Infof("Removing API group %v because it is time to stop serving it because it has no versions per APILifecycle.", groupName)
+			klog.V(1).InfoS("Removing API group because it is time to stop serving it because it has no versions per APILifecycle.", "groupName", groupName)
```
Comment (cici37) [inline pkg/controlplane/instance.go:604]: Same

File: pkg/controlplane/instance.go [hunk 6]
```diff
@@ -596,16 +601,17 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
...
 			continue
 		}
 
-		klog.V(1).Infof("Enabling API group %q.", groupName)
+		klog.V(1).InfoS("Enabling API group.", "groupName", groupName)
```
Comment (cici37) [inline pkg/controlplane/instance.go:608]: Same

File: pkg/controlplane/instance.go (hunk 3 above)
Comment (scottilee) [reply pkg/controlplane/instance.go:587]: Fixed.

File: pkg/controlplane/instance.go (hunk 4 above)
Comment (scottilee) [reply pkg/controlplane/instance.go:595]: Fixed.

File: pkg/controlplane/instance.go (hunk 5 above)
Comment (scottilee) [reply pkg/controlplane/instance.go:604]: Fixed.

File: pkg/controlplane/instance.go (hunk 6 above)
Comment (scottilee) [reply pkg/controlplane/instance.go:608]: Fixed.

**Comment by scottilee** [comment]
@cici37 Addressed your comments. Please let me know if there are any others.

File: pkg/controlplane/instance.go [hunk 7]
```diff
@@ -596,16 +601,17 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
 		// This is a spot above the construction of individual storage handlers so that no sig accidentally forgets to check.
 		resourceExpirationEvaluator.RemoveDeletedKinds(groupName, apiGroupInfo.Scheme, apiGroupInfo.VersionedResourcesStorageMap)
 		if len(apiGroupInfo.VersionedResourcesStorageMap) == 0 {
-			klog.V(1).Infof("Removing API group %v because it is time to stop serving it because it has no versions per APILifecycle.", groupName)
+			klog.V(1).InfoS("Removing API group because it is time to stop serving it because it has no versions per APILifecycle.", "apiGroup", groupName)
 			continue
 		}
 
-		klog.V(1).Infof("Enabling API group %q.", groupName)
+		klog.V(1).InfoS("Enabling API group.", "apiGroup", groupName)
```
Comment (yangjunmyfm192085) [inline pkg/controlplane/instance.go:608]:
```suggestion
		klog.V(1).InfoS("Enabling API group", "apiGroup", groupName)
```

File: pkg/controlplane/instance.go [hunk 8]
```diff
@@ -579,15 +584,15 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
...
+			klog.V(1).InfoS("Skipping disabled API group.", "apiGroup", groupName)
 			continue
 		}
 		apiGroupInfo, enabled, err := restStorageBuilder.NewRESTStorage(apiResourceConfigSource, restOptionsGetter)
 		if err != nil {
 			return fmt.Errorf("problem initializing API group %q : %v", groupName, err)
 		}
 		if !enabled {
-			klog.Warningf("API group %q is not enabled, skipping.", groupName)
+			klog.InfoS("API group is not enabled, skipping.", "apiGroup", groupName)
```
Comment (yangjunmyfm192085) [inline pkg/controlplane/instance.go:595]:
```suggestion
			klog.InfoS("API group is not enabled, skipping", "apiGroup", groupName)
```

File: pkg/controlplane/instance.go [hunk 9]
```diff
@@ -579,15 +584,15 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
 	for _, restStorageBuilder := range restStorageProviders {
 		groupName := restStorageBuilder.GroupName()
 		if !apiResourceConfigSource.AnyVersionForGroupEnabled(groupName) {
-			klog.V(1).Infof("Skipping disabled API group %q.", groupName)
+			klog.V(1).InfoS("Skipping disabled API group.", "apiGroup", groupName)
```
Comment (yangjunmyfm192085) [inline pkg/controlplane/instance.go:587]:
```suggestion
			klog.V(1).InfoS("Skipping disabled API group", "apiGroup", groupName)
```

File: pkg/controlplane/instance.go [hunk 10]
```diff
@@ -324,7 +329,7 @@ func (c *Config) Complete() CompletedConfig {
 		// but then that breaks the strict nestedness of ServiceType.
 		// Review post-v1
 		cfg.ExtraConfig.ServiceNodePortRange = kubeoptions.DefaultServiceNodePortRange
-		klog.Infof("Node port range unspecified. Defaulting to %v.", cfg.ExtraConfig.ServiceNodePortRange)
+		klog.InfoS("Node port range unspecified. Using default.", "ServiceNodePortRange", cfg.ExtraConfig.ServiceNodePortRange)
```
Comment (yangjunmyfm192085) [inline pkg/controlplane/instance.go:332]:
```suggestion
		klog.InfoS("Node port range unspecified. Using default", "ServiceNodePortRange", cfg.ExtraConfig.ServiceNodePortRange)
```

File: pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go [hunk 11]
```diff
@@ -159,10 +159,10 @@ func (c *Controller) syncConfigMap() error {
 	}
 
 	if equality.Semantic.DeepEqual(authConfigMap, originalAuthConfigMap) {
-		klog.V(5).Info("no changes to configmap")
+		klog.V(5).InfoS("No changes to ConfigMap")
 		return nil
 	}
-	klog.V(2).Infof("writing updated authentication info to  %s configmaps/%s", configMapNamespace, configMapName)
+	klog.V(2).InfoS("Writing updated authentication info to ConfigMaps", "configMapNamespace", configMapNamespace, "configMapName", configMapName)
```
Comment (yangjunmyfm192085) [inline pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go:165]: use klog.KRef?

File: pkg/controlplane/instance.go (hunk 10 above)
Comment (scottilee) [reply pkg/controlplane/instance.go:332]: Fixed.

File: pkg/controlplane/instance.go (hunk 8 above)
Comment (scottilee) [reply pkg/controlplane/instance.go:595]: Fixed.

File: pkg/controlplane/instance.go [hunk 12]
```diff
@@ -596,16 +601,17 @@ func (m *Instance) InstallAPIs(apiResourceConfigSource serverstorage.APIResource
 		// This is a spot above the construction of individual storage handlers so that no sig accidentally forgets to check.
 		resourceExpirationEvaluator.RemoveDeletedKinds(groupName, apiGroupInfo.Scheme, apiGroupInfo.VersionedResourcesStorageMap)
 		if len(apiGroupInfo.VersionedResourcesStorageMap) == 0 {
-			klog.V(1).Infof("Removing API group %v because it is time to stop serving it because it has no versions per APILifecycle.", groupName)
+			klog.V(1).InfoS("Removing API group because it is time to stop serving it because it has no versions per APILifecycle.", "apiGroup", groupName)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controlplane/instance.go:604]: Fixed.

File: pkg/controlplane/instance.go (hunk 7 above)
Comment (scottilee) [reply pkg/controlplane/instance.go:608]: Fixed.

File: pkg/controlplane/services.go [hunk 13]
```diff
@@ -34,7 +34,7 @@ import (
 func ServiceIPRange(passedServiceClusterIPRange net.IPNet) (net.IPNet, net.IP, error) {
 	serviceClusterIPRange := passedServiceClusterIPRange
 	if passedServiceClusterIPRange.IP == nil {
-		klog.Warningf("No CIDR for service cluster IPs specified. Default value which was %s is deprecated and will be removed in future releases. Please specify it using --service-cluster-ip-range on kube-apiserver.", kubeoptions.DefaultServiceIPCIDR.String())
+		klog.InfoS("No CIDR for service cluster IPs specified. Default value is deprecated and will be removed in future releases. Please specify it using --service-cluster-ip-range on kube-apiserver.", "defaultServiceIPCIDR", kubeoptions.DefaultServiceIPCIDR.String())
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controlplane/services.go:37]: Fixed.

File: pkg/controlplane/services.go [hunk 14]
```diff
@@ -48,7 +48,7 @@ func ServiceIPRange(passedServiceClusterIPRange net.IPNet) (net.IPNet, net.IP, e
 	if err != nil {
 		return net.IPNet{}, net.IP{}, err
 	}
-	klog.V(4).Infof("Setting service IP to %q (read-write).", apiServerServiceIP)
+	klog.V(4).InfoS("Setting service IP (read-write).", "apiServerServiceIP", apiServerServiceIP)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controlplane/services.go:51]: Fixed.

File: pkg/controlplane/tunneler/ssh.go [hunk 15]
```diff
@@ -192,9 +192,9 @@ func (c *SSHTunneler) nodesSyncLoop() {
 	// TODO (cjcullen) make this watch.
 	go wait.Until(func() {
 		addrs, err := c.getAddresses()
-		klog.V(4).Infof("Calling update w/ addrs: %v", addrs)
+		klog.V(4).InfoS("Calling update with addrs", "addrs", addrs)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/controlplane/tunneler/ssh.go:195]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085 Thanks for the review! I made all the corrections. Please let me know if there are any others.

**Comment by yangjunmyfm192085** [comment]
part duplicated with https://github.com/kubernetes/kubernetes/pull/100672

**Comment by scottilee** [comment]
Rebased.

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm

**Pull request closed without merging**